[Featurebranch] Check in for 2G events (EXPOSUREAPP-10632)

SamuraiKek opened the pull request:
Feature branch for 2G events

SamuraiKek committed:
UI preparations.

SamuraiKek committed:
Lint complaining.

SamuraiKek committed:
Fix test.

SamuraiKek committed:
Fix test.

SamuraiKek committed:
Changed enum to sealed class.

AlexanderAlferov committed:
Proper naming

AlexanderAlferov committed:
Admission badges

AlexanderAlferov committed:
Logic placeholder

AlexanderAlferov committed:
Updated data class with right certificates

LukasLechnerDev commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificates.kt:
```diff
@@ -14,4 +14,16 @@ data class PersonCertificates(
     }
+
+    sealed class AdmissionState(val primaryCertificate: CwaCovidCertificate) {
+        class Is2GPlusPCR(certificate2G: CwaCovidCertificate, certificateTest: CwaCovidCertificate) :
```
We changed it to `TwoG` and `TreeG`

LukasLechnerDev commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificates.kt:
```diff
@@ -14,4 +14,16 @@ data class PersonCertificates(
+
+        class Is2G(certificate2G: CwaCovidCertificate) : AdmissionState(certificate2G)
+        class Is3GWithPCR(certificateTest: CwaCovidCertificate) : AdmissionState(certificateTest)
+        class Is3GWithRAT(certificateTest: CwaCovidCertificate) : AdmissionState(certificateTest)
```
Added

SamuraiKek committed:
Updated admission state with data class.

LukasLechnerDev committed:
Make project compile again

LukasLechnerDev committed:
Fix PersonOverviewViewModelTest.kt

LukasLechnerDev committed:
Adjust AdmissionState so that primaryCertificate is now non-null

AlexanderAlferov committed:
G events person details (EXPOSUREAPP-10889) (#4485)

* Confirmed status card

* German FAQ link fix

* Test fix

SamuraiKek committed:
Fixed expecting non null object.

mtwalli committed:
Merge branch 'release/2.16.x' into feature/10632-check-in-for-2g-events

LukasLechnerDev committed:
Determine admission state (EXPOSUREAPP-10887) (#4493)

* Add extension function that determines the admission state of a person and add tests

* Fix Detekt Issues

* Fix PersonOverviewViewModelTest.kt

* Fix issue in PersonCertificatesExtensions.kt

* Fix test certificates with state EXPIRING_SOON to be also considered for admission state

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/items/PersonCertificateCard.kt:
```diff
@@ -25,32 +25,34 @@ class PersonCertificateCard(parent: ViewGroup) :
 
         val color = when {
-            curItem.certificate.isValid -> curItem.colorShade
+            curItem.admissionState.primaryCertificate!!.isValid -> curItem.colorShade
```
`!!` is not required

SamuraiKek replied:
This is not final, it's just until my PR will be merged into this one #4491

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/items/PersonCertificateCard.kt:
```diff
@@ -64,12 +66,13 @@ class PersonCertificateCard(parent: ViewGroup) :
     ) : PersonCertificatesItem, HasPayloadDiffer {
-        override val stableId: Long = certificate.personIdentifier.codeSHA256.hashCode().toLong()
+        override val stableId: Long =
+            admissionState.primaryCertificate!!.personIdentifier.codeSHA256.hashCode().toLong()
```
`!!` is not required

SamuraiKek replied:
Same here, will change with #4491

mtwalli replied:
it is not 😄

LukasLechnerDev committed:
Update dcc priorities (EXPOSUREAPP-10890) (#4504)

* Update dcc priorities: complete vaccinations and recent recoveries now have higher priority than recent test certificates

* Fix order of rules and adjust documentation

mtwalli committed:
Merge branch 'release/2.16.x' into feature/10632-check-in-for-2g-events

mtwalli committed:
Merge branch 'release/2.16.x' into feature/10632-check-in-for-2g-events

LukasLechnerDev committed:
Improve logic in ViewModels a bit (#4529)

mtwalli committed:
Merge branch 'release/2.16.x' into feature/10632-check-in-for-2g-events

mtwalli committed:
Merge branch 'release/2.16.x' into feature/10632-check-in-for-2g-events

SamuraiKek committed:
Check in for G events person overview UI changes (EXPOSUREAPP-10888) (#4491)

* Adapted certificate overview screen to support combined certs.

* Updated status badge text.

* Fixed lint & detekt.

* Deleted unused color.

* Fixed button color.

* Fixed dark mode color.

* Fixed no resource found error when cert was invalid.

* Minor UI adjustments

* Added screenshot for new G certificate.

* Added right test mocked certificate to UI test.

* Broken code down into smaller chunks for better readability.

* Updated FAQ link.

* Detekt.

Co-authored-by: Mohamed Metwalli <mohamed.metwalli@sap.com>

mtwalli committed:
Merge branch 'release/2.16.x' into feature/10632-check-in-for-2g-events

CV113 reviewed (commented):
@KathrinLuecke @hermesmar Hier sollten wir doch die Sätze umschreiben, oder? 
Also, anstatt von: Ihre Zertifikate ermöglichen Ihnen Zutritt für 3G Events...
--> Ihre Zertifikate erfüllen die 2G+ Regel
Und dann dementsprechend überall anpassen. Was meint ihr?

KathrinLuecke commented:
> @KathrinLuecke @hermesmar Hier sollten wir doch die Sätze umschreiben, oder? Also, anstatt von: Ihre Zertifikate ermöglichen Ihnen Zutritt für 3G Events... --> Ihre Zertifikate erfüllen die 2G+ Regel Und dann dementsprechend überall anpassen. Was meint ihr?

Gab's dazu eine Diskussion? Da war ich nicht involviert. Aber 3G und 2G+ ist ja nicht dasselbe, von daher verstehe ich diese Änderung jetzt nicht...?

jurajkusnier committed:
Merge branch 'release/2.16.x' into feature/10632-check-in-for-2g-events

SamuraiKek commented:
@CV113, @KathrinLuecke, is there a problem with the strings? Should we wait for some answer before merging this PR?

mtwalli committed:
Merge branch 'release/2.16.x' into feature/10632-check-in-for-2g-events

CV113 reviewed (commented):
I'm working on them right now

CV113 committed:
Update covid_certificate_strings.xml

--> Ihre Zertifikate ermöglichen Ihnen Zutritt für 2G Events INTO -->  Ihre Zertifikate erfüllen die ...G Regel.
We want to avoid the word events since we use it too often.

CV113 commented on Corona-Warn-App/src/main/res/values/covid_certificate_strings.xml:
```diff
@@ -393,4 +399,41 @@
+    <string name="confirmed_status_3g_plus_badge">"3G+"</string>
+
+    <!-- XTXT: Confirmed status card 2G body -->
+    <string name="confirmed_status_2g_body">"Ihre Zertifikate ermöglichen Ihnen Zutritt für 2G Events. Wenn Sie Ihren aktuellen Status vorweisen müssen, schließen sie diese Ansicht und zeigen Sie den QR-Code auf der Zertifikatsübersicht."</string>
```
Change:
 Ihre Zertifikate ermöglichen Ihnen Zutritt für 2 G Events 
 --> 
Ihre Zertifikate erfüllen die 2G Regel.

SamuraiKek replied:
Thanks, I see you have already pushed the new strings yourself. Can you please approve?

CV113 commented on Corona-Warn-App/src/main/res/values/covid_certificate_strings.xml:
```diff
@@ -393,4 +399,41 @@
+    <!-- XTXT: Confirmed status card 2G body -->
+    <string name="confirmed_status_2g_body">"Ihre Zertifikate ermöglichen Ihnen Zutritt für 2G Events. Wenn Sie Ihren aktuellen Status vorweisen müssen, schließen sie diese Ansicht und zeigen Sie den QR-Code auf der Zertifikatsübersicht."</string>
+    <!-- XTXT: Confirmed status card 2G+ body -->
+    <string name="confirmed_status_2g_plus_body">"Ihre Zertifikate ermöglichen Ihnen Zutritt für 2G+ Events. Wenn Sie Ihren aktuellen Status vorweisen müssen, schließen sie diese Ansicht und zeigen Sie den QR-Code auf der Zertifikatsübersicht."</string>
```
Change:
 Ihre Zertifikate ermöglichen Ihnen Zutritt für 2G+ Events 
 --> 
Ihre Zertifikate erfüllen die 2G+ Regel.

CV113 commented on Corona-Warn-App/src/main/res/values/covid_certificate_strings.xml:
```diff
@@ -393,4 +399,41 @@
+    <!-- XTXT: Confirmed status card 3G body -->
+    <string name="confirmed_status_3g_body">"Ihre Zertifikate ermöglichen Ihnen Zutritt für 3G Events. Wenn Sie Ihren aktuellen Status vorweisen müssen, schließen sie diese Ansicht und zeigen Sie den QR-Code auf der Zertifikatsübersicht."</string>
+    <!-- XTXT: Confirmed status card 3G+ body -->
+    <string name="confirmed_status_3g_plus_body">"Ihre Zertifikate ermöglichen Ihnen Zutritt für 3G+ Events. Wenn Sie Ihren aktuellen Status vorweisen müssen, schließen sie diese Ansicht und zeigen Sie den QR-Code auf der Zertifikatsübersicht."</string>
```
Change:
 Ihre Zertifikate ermöglichen Ihnen Zutritt für 3G+ Events 
 --> 
Ihre Zertifikate erfüllen die 3G+ Regel.

CV113 reviewed (changes requested):
UA requests changes

CV113 reviewed (approved):
UA approved
